querybook/webapp/components/QueryRunButton/QueryRunButton.tsx

```diff
@@ -287,29 +291,30 @@ const QueryLimitSelector: React.FC<{
     );
 };
 
-const TABLE_SAMPLING_CONFIG = PublicConfig.table_sampling ?? {
-    enabled: false,
-    sample_rates: [],
-    default_sample_rate: 0,
-};
-const sampleRateOptions = [{ label: 'none', value: 0 }].concat(
-    TABLE_SAMPLING_CONFIG.sample_rates.map((value) => ({
-        label: value + '%',
-        value,
-    }))
-);
-const DEFAULT_SAMPLE_RATE = TABLE_SAMPLING_CONFIG.default_sample_rate;
 const TableSamplingSelector: React.FC<{
-    sampleRate: number;
+    sampleRate: number | undefined;
     setSampleRate: (sampleRate: number) => void;
     tooltipPos: TooltipDirection;
     onTableSamplingInfoClick: () => void;
 }> = ({ sampleRate, setSampleRate, tooltipPos, onTableSamplingInfoClick }) => {
+    const sampleRateOptions = React.useMemo(getTableSamplingRateOptions, []);
+    const userDefaultTableSampleRate = useSelector(
+        (state: IStoreState) => state.user.computedSettings['table_sample_rate']
+    );
+
     React.useEffect(() => {
-        if (!sampleRateOptions.some((option) => option.value === sampleRate)) {
-            setSampleRate(DEFAULT_SAMPLE_RATE);
+        if (
+            sampleRate === undefined ||
+            !sampleRateOptions.some((option) => option.value === sampleRate)
+        ) {
+            setSampleRate(parseFloat(userDefaultTableSampleRate));
```

Comment:
do we need to watch out if userDefaultTableSampleRate is not in sampleRateOptions which could cause loop?

Reply:
in theory, the sample rate options of the user settings(including the user selected one) or here are both from the public config file, what you mentioned can not happen. But I can add a check to select the first one if it happens.